### ISSUE TITLE
`max-height` causes fullscreen videos on macOS Safari to have black borders

### ISSUE THREAD

**Issue by samdoshi** [tubearchivist/tubearchivist#264]
When viewing videos on macOS Safari in fullscreen 16:9 videos have black borders on all four sides.

Disabling the following 2 CSS rules in the Safari Developer Tools fixes the issue:

https://github.com/tubearchivist/tubearchivist/blob/e8e240c89bbad3aa48f6f40d51794ac0402ab5fd/tubearchivist/static/css/style.css#L719

https://github.com/tubearchivist/tubearchivist/blob/e8e240c89bbad3aa48f6f40d51794ac0402ab5fd/tubearchivist/static/css/style.css#L399

You might be able to use `:-webkit-full-screen`[^1] to override, if you're interested in making a change I can try and do a bit more research on it.

[^1]: https://caniuse.com/mdn-css_selectors_fullscreen

**Comment by bbilly1** [comment]
Thanks for looking into this. On what kind of screen size is that? Can you post a screenshot with and without the changes? I think that might be a Safari only issue.

**Comment by samdoshi** [comment]
Running v0.1.6:
<img width="1728" alt="before" src="https://user-images.githubusercontent.com/112490/176885202-40de8ca9-e95f-49de-aeda-a1bcb5601887.png">

After manually disabling both rules in Safari Developer Tools:
![after](https://user-images.githubusercontent.com/112490/176885454-263484e1-ac00-4f31-94e8-ef7b50857646.jpg)

Screen resolution is: 3456x2234 (that's the internal display on a 15" MacBook Pro), but the same issue occurs on an external 21:9 monitor.

No issues on Chrome. No issues on Safari on iPad either.

**Comment by bbilly1** [comment]
Thanks for the screenshots. I've been testing a bit. 

Just removing the *max-height* will create other problems, as a video is not guaranteed to be in horizontal orientation, for example shorts but also other sizes like square videos. So for these cases, *max-height* needs to be set, otherwise the video will scale based on the width value, e.g. `90%`. 

So your approach with the full screen pseudo class might be the right approach:
```css
video:-webkit-full-screen {
    max-height: unset;
}
```

Can you test this, if that is fixing the problem for Safari? As far as I can tell, this doesn't have any effect on Firefox and Chrome...

**Comment by samdoshi** [comment]
I needed to do the following:

```css
video:-webkit-full-screen {
    max-height: unset !important;
}
```

Which was placed at the very bottom of a local override of `style.css`.

I have no idea how to achieve the same without resorting to `!important` though.

A smaller issue (again only on Safari/Mac, not Chrome):

https://github.com/tubearchivist/tubearchivist/blob/6d59da4c8645cf0f7a89750711d7b9827ddcacc4/tubearchivist/static/css/style.css#L401

The `max-width` also causes issues on my 5120x2160 monitor, so perhaps this would be best:

```css
video:-webkit-full-screen {
    max-height: unset !important;
    max-width: unset !important;
}
```

**Comment by bbilly1** [comment]
This is now in v0.2.0. Don't hesitate to open any further issues with Safari, I don't have any access to that, so any improvements are very much welcome.

**Comment by samdoshi** [comment]
Thank you very much, just to report that it is fixed with `v0.2.0`.
